src/components/common/checkbox/CheckBox.tsx

```diff
@@ -0,0 +1,60 @@
+import clsx from 'clsx';
+import { forwardRef } from 'react';
+
+import { CheckBoxProps } from './CheckBox.types';
+import { CheckBoxIcon } from './CheckBoxIcon';
+
+import Label from '@/components/common/label/Label';
+
+const CheckBox = forwardRef<HTMLInputElement, CheckBoxProps>(function CheckBox(
+  { id, isIndeterminate, checked, disabled, className, labelText, ...restProps },
+  ref,
+) {
+  const baseStyles =
+    'relative outline-none radius-4xs inline-flex items-center justify-center border';
+
+  const enabledStyles =
+    checked || isIndeterminate
+      ? 'bg-feedback-information-dark border-border-trans-normal-dark'
+      : 'bg-object-static-inverse-hero-dark border-border-normal-dark';
+
+  const disabledStyles =
+    checked || isIndeterminate
+      ? 'bg-feedback-trans-information-dark border-border-trans-assistive-dark'
+      : 'bg-object-static-inverse-disabled-dark border-border-trans-assistive-dark';
+
+  return (
+    <div className={clsx('gap-2xs inline-flex items-center', className)}>
+      <div
+        className={clsx(
+          baseStyles,
+          disabled ? disabledStyles : enabledStyles,
+          'focus-within:before:shadow-focus-visible active:shadow-focus-visible duration-faster relative ease-(--motion-fluent) outline-none before:absolute before:inset-0 before:rounded-[inherit] hover:before:bg-[rgba(26,27,35,0.12)] active:before:bg-[rgba(26,27,35,0.12)]',
```

Comment:
`focus-within`으로 input이 focus되었을 때 div에 효과를 주신거군요
이렇게 하면 확실히 이벤트핸들러는 필요없겠네요 ㅎㅎ 
스토리북으로 확인했을 때도 키보드 탭, space로 체크 잘 되는 거 확인했습니다! 

이렇게 될 경우, 아래 문제가 있어요 
1. 마우스로 체크했을 때 outline이 계속 유지됨. 
2. outline이 border를 무시하고 감쌈

이 점들만 보완하면 좋을 것 같아요
 
한 번 방법을 생각해봤는데 `:has()` 의사 클래스를 사용하면 자식 요소의 상태에 따라서 스타일을 변경할 수 있어요 
```suggestion
          'has-focus-visible:shadow-focus-visible active:shadow-focus-visible duration-faster relative ease-(--motion-fluent) outline-none before:absolute before:inset-0 before:rounded-[inherit] hover:before:bg-[rgba(26,27,35,0.12)] active:before:bg-[rgba(26,27,35,0.12)]',
```
이렇게 하니까 키보드 탭, space 체크도 가능하고 마우스 사용 시에도 focus가 유지되지 않음을 확인했고 2번 문제인 border는 before를 제거해주면 됩니다

```suggestion
          'interaction-default-bold-inverse active:shadow-focus-visible duration-faster has-focus-visible:shadow-focus-visible ease-(--motion-fluent)',
```

색상이나 인터렉션 기본 스타일은 인터렉션 토큰과 동일해서 이렇게하면 좀 더 코드의 중복을 줄일 수 있는데 대신에 인터렉션 토큰의 `focus-visible:before:shadow-focus-visible`는 사용되지 않지만 불필요하게 들어가는 꼴이라 이 코드는 참고사항일 것 같습니다!

Reply:
`focus-within` 과 `active` 효과를 개별적으로 제어하는 방식도 시도해봤는데, 아무래도 제시해주신 자식 요소의 focus 상태에 따라 interaction을 부여해주는 방식이 가장 합리적인거 같네요.

[9d15603](https://github.com/JECT-Study/JECT-Official-WebSite-Client/pull/86/commits/9d156036dc5bebd5cc47ae1e8a16609e16251650) 에서 해당 사항 수정했습니다. 인터렉션 토큰을 사용할 경우, 사용되지 않는 focus 속성과 혼동이 있을 것 같아 기존 스타일을 유지했습니다~

Reply:
++) 개별 속성 제어 방식은 `:before` 제거를 통해 border 문제를 해결할 수 있었지만, 마우스로 체크했을 때 outline이 계속 유지되는 문제는 해결하기 까다롭더라구요!